compiler/rustc_mir_transform/src/const_prop.rs

```diff
@@ -143,8 +133,11 @@ impl ConstPropMachine<'_, '_> {
 
 impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for ConstPropMachine<'mir, 'tcx> {
     compile_time_machine!(<'mir, 'tcx>);
+
     const PANIC_ON_ALLOC_FAIL: bool = true; // all allocations are small (see `MAX_ALLOC_LIMIT`)
 
+    const POST_MONO_CHECKS: bool = false; // this MIR is still generic!
```

Comment:
Turns out this is not the entire reason, with const-generics we (try to) run regular CTFE on generic MIR as well.

It's just, we really don't need the post-mono checks here since we're not actually doing evaluation.